src/util/path/path.ts

```diff
@@ -651,9 +751,12 @@ export const getPathSegmentsInfo = (path: PathData): TPathSegmentsInfo[] => {
           current[5],
           current[6]
         );
-        tempInfo.iterator = iterator;
-        tempInfo.angleFinder = angleFinder;
-        tempInfo.length = pathIterator(iterator, x1, y1);
+        tempInfo = <TCurveInfo<'C'>>{
+          ...basicInfo,
```

Comment:
in my opinion there is no need to spread basicInfo every time.
basicInfo is a new object every loop, can just be added of new properties.

Reply:
I tried that, but it was causing similar issues to what I said [here](https://github.com/fabricjs/fabric.js/pull/8787#discussion_r1141161099). I agree it would be prettier but I couldn't make the types work nicely without a bunch of annoying casting/"as" stuff.

Reply:
I went ahead and changed it to non-spread since I realized I was already doing a cast, derp. It's actually less lines than before, so I'm going to resolve this convo.